File: scripts/upsert_from_feed.ts
import SupabaseClientWrapper from "../src/tools/supabaseClient";

// Minimal Apartment shape used by this script to avoid importing internal types
type Apartment = {
  external_id: string;
  source?: string;
  title?: string;
  address?: string;
  city?: string;
  state?: string;
  zip_code?: string;
  rent_price?: number;
  bedrooms?: number;
  bathrooms?: number;
  square_feet?: number;
  amenities?: string[];
  images?: string[];
};
import * as process from "node:process";

function scrapeApartments(): Apartment[] {
  // Replace this with real scraping logic or import an existing scraper.
  return [
    {
      external_id: "ts-example-1",
      source: "ts_example",
      title: "TS 1BR",
      address: "1 TS Way",
      city: "Metropolis",
      state: "NY",
      zip_code: "10001",
      rent_price: 1500,
      bedrooms: 1,
      bathrooms: 1,
      square_feet: 650,
      amenities: ["elevator", "gym"],
    },
  ];
}

async function main() {
  const dryRun = !!process.env.DRY_RUN;
  const supa = new SupabaseClientWrapper();
  const apartments = scrapeApartments();

  for (const apt of apartments) {
    if (dryRun) {
      console.log("DRY_RUN payload:", apt);
      continue;
    }
    const res = await supa.upsertApartment(apt);
    console.log("Upserted:", res);
  }

  if (!dryRun) {
    const deactivated = await supa.deactivateOldListings("ts_example", 14);
    console.log("Deactivated:", deactivated);
  }
}

if (typeof require !== "undefined" && require.main === module) {
  main().catch((e) => {
    console.error(e);
    process.exit(1);
  });
}

File: scripts/upsert_from_feed.py
from database.supabase_client import SupabaseClient


def scrape_apartments():
    """Placeholder scraper function. Replace with real scraping logic.
    Returns a list of dicts representing apartments.
    """
    # Example data; replace with your actual scraping output
    return [
        {
            "external_id": "example-1",
            "source": "example_feed",
            "title": "1BR Apartment",
            "address": "123 Main St",
            "city": "Springfield",
            "state": "IL",
            "zip_code": "62701",
            "rent_price": 1200,
            "bedrooms": 1,
            "bathrooms": 1,
            "square_feet": 600,
            "amenities": ["dishwasher", "air_conditioning"],
            "images": [],
        }
    ]


def main():
    supabase = SupabaseClient()

    apartments = scrape_apartments()

    for apt in apartments:
        row = supabase.upsert_apartment(apt)
        print("Upserted:", row)

    # Clean up old listings
    updated = supabase.deactivate_old_listings("example_feed", cutoff_days=14)
    print(f"Deactivated {updated} old listings")


if __name__ == "__main__":
    main()
